### DISCUSSION TITLE
Compiling and ugrading en ESP32 - Program flash size is larger than real flash size

### DISCUSSION THREAD

**Discussion by SimonFili** [arendst/Tasmota#19319] [Support request] [answered]
Hi all,

So here's what I'm trying to accomplish:
1) I bought a Rocket Controller ESP32R4 box.  This box comes with:
 - Version : 12.1.1.6 (tasmota)
 - Core/SDK Version : 2_0_5/v4.4.3.4
- ![image](https://github.com/arendst/Tasmota/assets/7669537/daa47ccc-9b39-49f6-992e-284c0404cced)
- The compiled version support the PCF8574 ( I see the button to configure it)

2) I want to update the firmware to support some EZO chips from Atlas Scientific

So I went to the compiler on gitpod and added (removed comment) on the EZO chip I need and the lines for the PCF8574, compiled it and uploaded the firmware to update it.  I compiled with ESP32 (tasmot32 in platformio_overdide). I assumed the ESP32-D0WD is a normal ESP32 (right?)

When I upload to update, I get this message:
Program flash size is larger than real flash size

After a few reading on the forum, some suggested to re-partition, so following what I found here : https://tasmota.github.io/docs/Tasmota-Application/
I added the partition wizard to the device and re-partition it to the max.

Tried again, it did not work. Same error.

What steps am I missing? ( I'm sure is obvious to some ! )
Recommendations? 

I hope I gave enough information.  Don't hesitate to ask for more.

**Comment by Jason2866** [comment]
Convert to safeboot partition scheme. See https://github.com/arendst/Tasmota/discussions/18983

**Reply by SimonFili** [reply, marked as answer]
Alright, upload the partition wizard again, I have this:
![image](https://github.com/arendst/Tasmota/assets/7669537/2278db01-6a7f-4b66-a546-4b393ea3e8f4)

Since I already did another steps before, do I simply click on start migration?  I assume yes, since there's no other option.

**Reply by SimonFili** [reply]
So I ran the migration using this partition tool: https://tasmota.github.io/docs/Tasmota-Application/#partition-management

Then I uploaded my custom firmware with EZO and PCF8574, works fine!

Thanks

**Comment by SimonFili** [comment]
Here's the output when this process is ongoing
![image](https://github.com/arendst/Tasmota/assets/7669537/675df560-2c68-4913-b91a-d641c088ad40)
